asset-transfer-private-data/chaincode-go/privateAssetQueries.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/hyperledger/fabric-contract-api-go/contractapi"
+)
+
+
+// read asset information from collection
+func (s *SmartContract) ReadAsset(ctx contractapi.TransactionContextInterface, assetID string) (*Asset, error) {
+
+	assetJSON, err := ctx.GetStub().GetPrivateData("assetCollection", assetID) //get the asset from chaincode state
+	if err != nil {
+			return nil, fmt.Errorf("failed to read from asset %s", err.Error())
+		}
+		if assetJSON == nil {
+			return nil, fmt.Errorf("%s does not exist", assetID)
+		}
+
+		asset := new(Asset)
+	_ = json.Unmarshal(assetJSON, asset)
+
+	return asset, nil
+
+}
+
+// read asset private details in organization specific collection
+func (s *SmartContract) ReadAssetPrivateDetails(ctx contractapi.TransactionContextInterface, collection string, assetID string) (*AssetPrivateDetails, error) {
+
+	assetDetailsJSON, err := ctx.GetStub().GetPrivateData(collection, assetID) //get the asset from chaincode state
+		if err != nil {
+			return nil, fmt.Errorf("failed to read from asset details %s", err.Error())
+		}
+		if assetDetailsJSON == nil {
+			return nil, fmt.Errorf("%s does not exist", assetID)
+		}
+
+		assetDetails := new(AssetPrivateDetails)
+	_ = json.Unmarshal(assetDetailsJSON, assetDetails)
+
+	return assetDetails, nil
+}
+
+// ===========================================================================================
+// GetAssetByRange performs a range query based on the start and end keys provided.
+
+// Read-only function results are not typically submitted to ordering. If the read-only
+// results are submitted to ordering, or if the query is used in an update transaction
+// and submitted to ordering, then the committing peers will re-execute to guarantee that
+// result sets are stable between endorsement time and commit time. The transaction is
+// invalidated by the committing peers if the result set has changed between endorsement
+// time and commit time.
+// Therefore, range queries are a safe option for performing update transactions based on query results.
```

Comment:
I missed this in marbles02_private... the comment is not correct for private data... you can't do range queries in update transactions with private data, since not all peers can validate the result set at validation time.
We can just remove this comment.

Reply:
done.